Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs

```diff
@@ -1426,4 +1454,22 @@ getSelectedEstimatesObject :: String -> Maybe ChooseVehicle.Config
 getSelectedEstimatesObject dummy =
   case runExcept (decodeJSON (getValueToLocalStore ESTIMATE_DATA) :: _ ChooseVehicle.Config) of
     Right res -> Just res
-    Left err -> Nothing
\ No newline at end of file
+    Left err -> Nothing
+
+getChatSuggestions :: ST.HomeScreenState -> Array String
+getChatSuggestions state = do 
+  let didDriverMessage = (HU.didDriverMessage FunctionCall)
+      canShowSuggestions = case (DA.last state.data.messages) of 
+                          Just value -> ((not $ value.sentBy == "Customer") || not didDriverMessage)
+                          Nothing -> true
+      isAtPickup = (metersToKm state.data.driverInfoCardState.distance state) == getString AT_PICKUP
+  if (DA.null state.data.suggestionsList) && canShowSuggestions && state.props.canSendSuggestion then
+    if didDriverMessage && (not $ DA.null state.data.messages) then
+      if isAtPickup then getSuggestionsfromKey "customerDefaultAP" else getSuggestionsfromKey "customerDefaultBP"
+    else 
+    if isAtPickup then getSuggestionsfromKey "customerInitialAP" else do
+        let hideInitial = (not (DA.null state.data.messages)) && (not didDriverMessage)
+        if (DA.null state.data.messages) && (EHC.getExpiryTime state.data.driverInfoCardState.createdAt true) > 30 then getSuggestionsfromKey "customerInitialBP3"
+        else if hideInitial then getSuggestionsfromKey "customerInitialBP2"
+        else getSuggestionsfromKey "customerInitialBP1"
+  else state.data.suggestionsList
```

Comment:
Please rename this Home Screen state variable to `chatSuggestionList`

Reply:
These are not only for chat